event.go

```diff
@@ -59,6 +61,9 @@ func (e HTTPRequestEvent) Key() string {
 type DNSRequestEvent struct {
 	// The DNS question received.
 	Question dns.Question
+	// The User-Agent header from the request, may be empty
+	// if the request was not over DoH.
+	UserAgent string
```

Comment:
I was surprised to see User-Agent showing up for DNS requests, but this makes sense!

As an FYI, I believe the DoH client in Boulder simply uses the Go default user-agent. We don't set the user-agent based on the VA's user-agent config, largely because the DoH requests are always internal, and we don't particularly care about Unbound seeing an informative user-agent, since we know where all our traffic is coming from.